src/librustc/mir/mod.rs

```diff
@@ -1080,8 +1080,18 @@ pub enum TerminatorKind<'tcx> {
     /// Indicates a terminator that can never be reached.
     Unreachable,
 
-    /// Drop the `Place`.
-    Drop { location: Place<'tcx>, target: BasicBlock, unwind: Option<BasicBlock> },
+    /// Drop the `Place`, possibly conditioned on a flag being true.
+    Drop {
+        location: Place<'tcx>,
+        /// Whether to drop the value.
+        ///
+        /// Before drop elaboration this is always `None. After drop elaboration
+        /// If this is `None` then the drop is unconditional, otherwise the drop
+        /// is only evaluated when the flag is true.
+        flag: Option<Place<'tcx>>,
```

Comment:
Why is this a place, when all we do is read from it (so `Operand` would be enough)?